### PR TITLE
London Class 7 - Aisha Khan - JavaScript-Core-1-Coursework- Week 3

### PR DESCRIPTION
**Pull request by aishak05**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

File: 2-mandatory/3-space-colonies.js [hunk 1]
```diff
@@ -15,7 +15,11 @@
   
 */
 
-function getSettlers() {}
+function getSettlers(arr) {
+  const Family = arr.filter((ele) => ele.includes("family"));
```
Comment (ademalagoz) [inline 2-mandatory/3-space-colonies.js:19]: Hi Aisha. Great work. I only suggest that as mentioned below link we supposed to use 'lowerCamelCasing' method for names at this and next challenges.
https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/JavaScript

File: 1-exercises/J-string-substring/exercise3.js [hunk 2]
```diff
@@ -7,8 +7,12 @@
 */
 
 let statement = "I do not like programming";
+let firstHalf = statement.substring(0,5);
```
Comment (Hozan94) [inline 1-exercises/J-string-substring/exercise3.js:10]: Hi Aisha, great work here, If I can suggest something is maybe you can write this method shorter like below, but overall well done:
`let statement = "I like programming and dogs";

statement = statement.substring(0,19);

console.log(statement);`

File: 2-mandatory/1-oxygen-levels.js [hunk 3]
```diff
@@ -11,7 +11,11 @@
     Some string methods that might help you here are .replace() and .substring(). 
 */
 
-function findSafeOxygenLevel() {}
+function findSafeOxygenLevel(arr) {
+   const newArr = arr.map((ele) => ele.replace("%", ""));
+   return newArr.find((ele) => ele > 19.5 && ele < 23.5) + "%";
```
Comment (Hozan94) [inline 2-mandatory/1-oxygen-levels.js:16]: Thanks to you, now I know when using strings in comparision operator they are converted to numbers.

File: 2-mandatory/6-lane-names.js [hunk 4]
```diff
@@ -6,7 +6,12 @@
   HINT: string and array methods that could be helpful (indexOf, filter)
 */
 
-function getLanes() {}
+function getLanes(arr) {
+  return arr.filter(arr => {
```
Comment (Hozan94) [inline 2-mandatory/6-lane-names.js:10]: If I was I would change the name of the parameter "arr" inside the filter() method to something else, maybe like 'newArr', cause it looks like it takes the first 'arr' as an argument but it is not .

**Review by Hozan94 — COMMENTED** [review]
Hi @aishak05 , Well done on completing the assignment, your coding is very good, keep it up.
I also left few suggestions to consider.
Many thanks

**Review by yunusfirat — COMMENTED** [review]
Could you please check your fizzbuzz solution again ? 
please make sure that you have obtained expected solution. 
hint: you should check the number that can be divided 3 and 5 first.

**Pull request closed without merging**